### PR TITLE
Making blood less frustrating

### PR DESCRIPTION
**Pull request by Kush1Push1**
## About The Pull Request

Two changes.
```
1. If a splatter of blood isn't bloody enough, the character will not get bloody shoes.

2. Characters with light steps trait won't get bloody clothes if they step on a bloody tile without shoes.

3. added: Blood dries over time. Usually, 3 or 6 minutes. If a splatter is big enough, it can stay wet up to 15 minutes.
```

## Why It's Good For The Game

The entire crew won't have bloody shoes anymore.
Less time wasted looking for showers or sinks to wash their shoes after stepping on a near invisible blood decal.
Characters without shoes (or who don't want to get their clothes bloody after taking them off) will get more use from a near useless trait.

## Changelog
:cl:
qol: Blood splatters only make your shoes or feet bloody if the tile is bloody enough,
qol: Light steps trait prevents clothing from being covered in blood if you step on a bloody tile with your shoes off.
add: Blood dries over time. Usually, 3 or 6 minutes. If a splatter is big enough, it can stay wet up to 15 minutes.
/:cl:

(grammaticly impoved by @maxymax13)

### CODE REVIEW

**Comment by vpellen** [comment]
Thank you I'm massively autistic about keeping my feet clean
Is it possible to just have blood "dry out" over time, at which point it stops leaving prints altogether?

**Comment by maxymax13** [comment]
> Is it possible to just have blood "dry out" over time, at which point it stops leaving prints altogether?

that would be great

**Comment by maxymax13** [comment]
## About The Pull Request

Two changes.

    1. If a splatter of blood isn't bloody enough, the character will not get bloody shoes.

    2. Characters with light steps trait won't get bloody clothes if they step on a bloody tile without shoes.


( :) )
## Why It's Good For The Game

The entire crew won't have bloody shoes anymore.
Less time wasted looking for showers or sinks to wash their shoes after stepping on a near invisible blood decal.
Characters without shoes (or who don't want to get their clothes bloody after taking them off) will get more use from a near useless trait.
## Changelog

🆑
tweak: Blood splatters only make your shoes or feet bloody if the tile is bloody enough,
tweak: Light steps trait prevents clothing from being covered in blood if you step on a bloody tile with your shoes off.
/🆑

**Comment by Farquaar** [comment]
This is good. When everybody has bloody shoes, no one has bloody shoes.
Maybe now, having bloody shoes will actually be uncommon enough to make it slightly suspicious.

**Comment by Kush1Push1** [comment]
@maxymax13 Thank you!

**Comment by Fox-McCloud** [comment]
you could make it so that there's a timer created for blood---after a certain amount of time the timer pops and the blood dries, therefore preventing it from making shoes icky.

File: code/modules/detectivework/detective_work.dm [hunk 1]
```diff
@@ -81,6 +81,8 @@
 	return TRUE //we bloodied the floor
 
 /mob/living/carbon/human/add_blood_DNA(list/blood_dna, list/datum/disease/diseases)
+	if(HAS_TRAIT(src, TRAIT_LIGHT_STEP)) //the character is agile enough to don't mess their clothing and hands just from one blood splatter at floor
+		return TRUE
```
Comment (coiax) [inline code/modules/detectivework/detective_work.dm:85]: So this proc, `add_blood_DNA` is _only_ called by walking over a blood splatter, and no other times?

Or does this change give LIGHT_STEP trait holders the ability to not get bloody anymore?

Comment (Kush1Push1) [reply code/modules/detectivework/detective_work.dm]: Bloodysoles.dm is the only file that has references to `add_blood_DNA`, yeah.
But I still fear something could be overlocked, so I moved the code highter

**Comment by Kush1Push1** [comment]
@coiax Could you review again?

**Comment by Kush1Push1** [comment]
Uh, right, I added things
Updating the description
Added:
added: Blood dryes over time. Usually, 3 or 6 minutes. If a splatter is big enough, it can stay wet up to 15 minutes.

**Comment by Kush1Push1** [comment]
I also tested how it works with virus. You still get infection without shoes (as intended) if have the trait.

**Comment by Kush1Push1** [comment]
Some screenshots
<details>

dreamseeker_nP7K1ENX9U
dreamseeker_WIJC1gGncY
dreamseeker_ll1dLd0gas

</details>

**Comment by Kush1Push1** [comment]
Oof .-.'

**Comment by MrMelbert** [comment]
just a warning, `tweak` tag in changelogs doesn't exist anymore, the tag is now QOL (and the tag for features is `add`
```
qol: Blood splatters only make your shoes or feet bloody if the tile is bloody enough,
qol: Light steps trait prevents clothing from being covered in blood if you step on a bloody tile with your shoes off.
add: Blood dryes over time. Usually, 3 or 6 minutes. If a splatter is big enough, it can stay wet up to 15 minutes.
```

**Comment by Kush1Push1** [comment]
> just a warning, `tweak` tag in changelogs doesn't exist anymore, the tag is now QOL (and the tag for features is `add`, not added)
> 
> ```
> qol: Blood splatters only make your shoes or feet bloody if the tile is bloody enough,
> qol: Light steps trait prevents clothing from being covered in blood if you step on a bloody tile with your shoes off.
> add: Blood dries over time. Usually, 3 or 6 minutes. If a splatter is big enough, it can stay wet up to 15 minutes.
> ```

Thank you!

**Comment by optimumtact** [comment]
I'm not a huge fan of the use of timers because there can be a lot of blood decals on a bad round.

do we have an existing subsystem these could process on?

**Comment by Kush1Push1** [comment]
> I'm not a huge fan of the use of timers because there can be a lot of blood decals on a bad round.
> 
> do we have an existing subsystem these could process on?

Replaced with process() on obj subsystem.

**Pull request merged**